.github/workflows/pull-request-main.yml

```diff
@@ -0,0 +1,127 @@
+name: Pull request (main) CI tests
+
+on:
+  push:
+    branches:
+      - main
+      # TODO "gh-actions" branch included to facilitate development
+      # we could also consider simply going myfork:main -> upstream:main for CI PRs
+      - gh-actions
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      git-ref:
+        description: Git hash (optional)
+        required: false
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  run-tests:
+    name: Install and test (py=${{ matrix.python-version }}, os=${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      # if fail-fast == true (the default), jobs for the remaining values in the matrix are cancelled
+      fail-fast: false
+      matrix:
+        python-version:
+          - '3.6'
+          - '3.7'
+        os:
+          - ubuntu-18.04
+          - windows-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install IDAES with dependencies
+        run: |
+          python -m pip install --progress-bar off -r requirements.txt
+      - name: Display IDAES version
+        run: |
+          idaes --version
+      - name: Install IDAES extensions
+        run: |
+          idaes get-extensions --verbose
+          # TODO: these commands are Unix-shell specific 
+          # and will need to be changed if we want to test on CMD.exe/Powershell
+          find $(idaes data-directory) -ls
```

Comment:
Is this comment out of date, as GHA makes the bash shell available on Windows?

Reply:
Here I was making a note that having `find` requires a Unix shell, i.e., they will work on Windows as long as `shell: bash` is set or, conversely, they will fail on Windows with the other available shells, including the case where `shell` is not specified, since (as I understand) it's the only available OS where the default shell is not a Unix shell.